docs/src/main/asciidoc/native.adoc

```diff
@@ -0,0 +1,107 @@
+
+https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#cloud-monitoring[Cloud Monitoring]
+
+https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#spring-data-cloud-spanner[Spring Data Cloud Spanner] (See details in <<detail_guides>> )
+
+https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#spring-data-cloud-datastore[Spring Data Cloud Datastore] (See details in <<detail_guides>> )
+
+https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#spring-data-cloud-firestore[Spring Data Cloud Firestore] (See details in <<detail_guides>> )
+
+https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#cloud-vision[Cloud Vision]
+
+https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#secret-manager[Cloud Secret Manager]
+
+https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#google-cloud-key-management-service[Cloud KMS]
+
+=== Usage
+
+This section provides a quick guide to build native image for Spring Boot Application with Google Cloud integrations.
+For more details, please refer to Spring boot guide to https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#native-image.developing-your-first-application[Developing Your First GraalVM Native Application]
+
+==== Paketo Buildpacks
+
+Doesn't require GraalVM to be installed on the local environment.
+Needs docker to be installed and the following
+
+1. spring-boot-starter-parent
+
+[source]
+----
+<parent>
+<groupId>org.springframework.boot</groupId>
+<artifactId>spring-boot-starter-parent</artifactId>
+</parent>
+----
+
+2. native-maven-plugin
+
+[source]
+----
+<plugin>
+<groupId>org.graalvm.buildtools</groupId>
+<artifactId>native-maven-plugin</artifactId>
+</plugin>
+----
+
+3. Call `mvn -Pnative spring-boot:build-image`
+
+==== Native Build Plugins
+
+Needs GraalVM to be installed on the local environment and the following:
+
+1. `org.springframework.boot:spring-boot-starter-parent`.
+No special dependencies for native image support.
+It is all baked into the spring boot dependency.
+This dependency provides the `native` profile.
+2. The `spring-boot-maven-plugin` - provides Spring Boot support in Maven and inherits a lot of default configurations from the spring-boot-starter-parent.
+
+[source]
+----
+<plugin>
+<groupId>org.springframework.boot</groupId>
+<artifactId>spring-boot-maven-plugin</artifactId>
+</plugin>
+----
+
+3. `org.graalvm.buildtools: native-maven-plugin`.
+The spring boot maven plugin uses this to activate native image compilation.
+4. Call `mvn -Pnative native:compile`
+
+=== Guides Specific to a Module [[detail_guides]]
+
+==== Datastore
+
+To use `DatastoreTemplate` methods on Domain types not managed by `DatastoreRepository`, you need to register runtime hints for these domain POJOs in your application. For more details, refer to example
```

Comment:
I thought these were `@Entity` classes?

Reply:
Trying to align with our documentation wording here. Regarding object mapping, we state in documentation: "Spring Data Cloud Datastore allows you to map domain POJOs to Cloud Datastore kinds and entities via annotations:" ([code](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/1ea0cfb5769a132af43440d96eb7d23b063d3d03/docs/src/main/asciidoc/datastore.adoc?plain=1#L112))